### PR TITLE
Update modules_and_classes.rst

### PR DESCRIPTION
**Pull request by aaronayres35**
In some ways this is a follow up to #637 and as such is part of #586 

This PR cleans up formatting and adds to the content of the "Commonly Used Modules and Classes" page of the documentation.

### CODE REVIEW

File: docs/source/user_manual/modules_and_classes.rst [hunk 1]
```diff
@@ -237,19 +283,48 @@ first in Chaco, and then moved into Enable.
...
+* :class:`~.TextBoxOverlay`
+* :class:`~.ToolTip`
+* :class:`~.StatusLayer`
+* :class:`~.CoordinateLineOverlay`
+* :class:`~.DataBox`
+* :class:`~.ImageInspectorColorbarOverlay`
+* :class:`~.RangeSelectionOverlay`
 
+Miscellaneous
+-----------------------------------------------------------------------------
```
Comment (rahulporuri) [inline docs/source/user_manual/modules_and_classes.rst:330]: let's remove the empty misc section.

File: docs/source/user_manual/modules_and_classes.rst [hunk 2]
```diff
@@ -237,19 +283,48 @@ first in Chaco, and then moved into Enable.
   * :class:`~.ScatterInspector`
   * :class:`~.SelectableLegend`
 
+* :class:`~.SimpleInspectorTool`
 * :class:`enable.tools.traits_tool.TraitsTool`
+* :class:`~chaco.tools.traits_tool.TraitsTool`
```
Comment (rahulporuri) [inline docs/source/user_manual/modules_and_classes.rst:288]: tangential question - these two look similar but im not sure what their differences are - and why

File: docs/source/user_manual/modules_and_classes.rst [hunk 3]
```diff
@@ -197,36 +238,41 @@ first in Chaco, and then moved into Enable.
...
-  * :class:`~.LegendTool`
-  * :class:`~.MoveTool`
-
 * :class:`~.DrawPointsTool`
 * :class:`~.HighlightTool`
 * :class:`enable.tools.hover_tool.HoverTool`
 * :class:`~.ImageInspectorTool`
 * :class:`~.LineInspector`
-* :class:`~.PanTool`
+* :class:`chaco.tools.pan_tool.PanTool`
```
Comment (rahulporuri) [inline docs/source/user_manual/modules_and_classes.rst:275]: tangential question - this and `pan_tool2` look similar but im not sure what their differences are - and why

**Review by rahulporuri — APPROVED** [review]
LGTM

File: docs/source/user_manual/modules_and_classes.rst (hunk 3 above)
Comment (aaronayres35) [reply docs/source/user_manual/modules_and_classes.rst:275]: I agree, I've moved #438 into the project for the quarter

**Pull request merged**